### PR TITLE
IS-1230: getPReps returns error with high end ranking.

### PR DESCRIPTION
**Pull request by cow-hs**
Invalid case
* zero startRanking
* zero endRanking
* reverse

else:
 normal case

fixes icon-project/icon-rpc-server#147

### CODE REVIEW

File: iconservice/prep/engine.py [hunk 1]
```diff
@@ -912,11 +912,11 @@ def handle_get_preps(self,
 
         if startRanking is None:
             startRanking = 1
-        if endRanking is None:
+        if endRanking is None or endRanking > prep_count:
             endRanking = prep_count
 
         if prep_count > 0:
-            if not 1 <= startRanking <= endRanking:
+            if startRanking < 1 or endRanking < 1 or startRanking - 1 >= endRanking:
```
Comment (inwonkim) [inline iconservice/prep/engine.py:919]: How about `startRanking < 1 or startRanking > endRanking` ?

Comment (leeheonseung) [reply iconservice/prep/engine.py:919]: good suggestion

File: tests/integrate_test/iiss/prevote/test_prep.py [hunk 2]
```diff
@@ -633,3 +633,147 @@ def check_query_via_icx_sendtransaction(self, method: str, params: dict, expecte
                                        func_name=method,
                                        params=params)
         return self.process_confirm_block_tx([tx], expected_status=expected_status)
+
+    def test_get_preps_normal(self):
```
Comment (goldworm-icon) [inline tests/integrate_test/iiss/prevote/test_prep.py:637]: I want to add additional test-cases
1. startRanking or endRanking are less than zero
2. startRanking is larger than # of PReps

**Review by goldworm-icon — CHANGES_REQUESTED** [review]

File: tests/integrate_test/iiss/prevote/test_prep.py (hunk 2 above)
Comment (leeheonseung) [reply tests/integrate_test/iiss/prevote/test_prep.py:637]: append test case.

pass case
* normal
* over endRanking than prep count

else:
* reverse
* overflow
* zero
* negative

**Review by goldworm-icon — APPROVED** [review]

**Pull request merged**